pkg/kf/commands/apps/integration_test.go

```diff
@@ -95,6 +100,37 @@ func TestIntegration_Push_docker(t *testing.T) {
 	})
 }
 
+// TestIntegration_Push pushes the echo app, lists it to ensure it can find a
+// domain, uses the proxy command and then posts to it. It finally deletes the
+// app.
+func TestIntegration_Push_dockerfile(t *testing.T) {
+	t.Parallel()
+	checkClusterStatus(t)
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		currentTime := time.Now().UnixNano()
+		appName := fmt.Sprintf("integration-dockerfile-%d", currentTime)
+		appPath := filepath.Join(RootDir(ctx, t), "samples", "apps", "dockerfile")
+
+		// Create a custom manifest file for this test.
+		newManifestFile, manifestCleanup, err := copyManifest(appName, appPath, currentTime)
+		AssertNil(t, "app manifest copy error", err)
+		defer manifestCleanup()
+
+		kf.Push(ctx, appName, "--path", appPath, "--manifest", newManifestFile)
+		defer kf.Delete(ctx, appName)
+
+		checkApp(ctx, t, kf, appName, []string{ExpectedAddr(appName, "")}, 8089, func(ctx context.Context, t *testing.T, addr string) {
```

Comment:
nit: could we make this Dockerfile app simply build the helloworld app so that we can just use `checkHelloWorldApp`?

Reply:
Possibly, I was using the different strings in the apps to prove that they were distinct instances because the HelloWorldApp was incorrectly validating correctly against echo before. 

Maybe it would be best to have an app whose name is returned when the body is requested via `VCAP_APPLICATION`?

Reply:
I like that idea.